### PR TITLE
Migration to Django 2.2

### PR DESCRIPTION
**Pull request by Ram81**
### Description

This PR adds:

- [x] Upgrade `django==1.11.3` to `django=2.2.8`
- [x] Add respective models changes (specify `on_delete` for `ForeignKey` fields, refer [this](https://docs.djangoproject.com/en/3.1/ref/models/fields/))
- [x] Change `user.is_anonymous()` to `user.is_anonymous`, refer [this](https://docs.djangoproject.com/en/2.2/ref/contrib/auth/)
- [x] Upgrade `drf-yasg` `pyyaml` `django-allauth` `djangorestframework` `django-silk` `psycopg2`
- [x] Fix tests to use`django.urls.reverse_lazy` imports.
- [x] Fix `urlsafe_base64_encode` usage
- [x] Add `app_name` in `urls.py` for app url resolution.

### CODE REVIEW

File: apps/challenges/views.py [hunk 1]
```diff
@@ -627,6 +627,8 @@ def challenge_phase_detail(request, challenge_pk, pk):
         return Response(response_data, status=status.HTTP_406_NOT_ACCEPTABLE)
 
     if request.method == "GET":
+        print("In request")
```
Comment (RishabhJain2018) [inline apps/challenges/views.py:630]: Please remove the print statement.

File: apps/challenges/views.py [hunk 2]
```diff
@@ -627,6 +627,8 @@ def challenge_phase_detail(request, challenge_pk, pk):
...
+        print(request.data)
```
Comment (RishabhJain2018) [inline apps/challenges/views.py:631]: Please remove the print statement.

File: apps/web/views.py [hunk 3]
```diff
@@ -29,7 +29,7 @@ def home(request, template_name="index.html"):
     return render(request, template_name)
 
 
-def page_not_found(request):
+def page_not_found(request, exception):
```
Comment (RishabhJain2018) [inline apps/web/views.py:32]: Why are we passing `exception` as a parameter when we are not using it?

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]

File: apps/web/views.py (hunk 3 above)
Comment (Ram81) [reply apps/web/views.py:32]: @RishabhJain2018 please refer [this](https://docs.djangoproject.com/en/3.1/ref/views/). In django 2.0 `page_not_found` function has exception as mandatory param.

**Comment by Ram81** [comment]
@RishabhJain2018 I've made the changes

**Review by RishabhJain2018 — CHANGES_REQUESTED** [review]
@Ram81 Since you changed the models, we also need to have migration files right?

**Comment by Ram81** [comment]
@RishabhJain2018 no, I ran `makemigrations` and it didn't detect any changes. By default value of `on_delete` was `models.CASCADE` which is why we don't need any migrations for these changes.

**Review by RishabhJain2018 — COMMENTED** [review]
@Ram81 I can merge this one, right?

**Comment by Ram81** [comment]
@RishabhJain2018  yes

**Review by RishabhJain2018 — APPROVED** [review]

**Pull request merged**